libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.cpp

```diff
@@ -1357,93 +1357,85 @@ int HTTPClient::writeToStreamDataBlock(Stream * stream, int size)
     // create buffer for read
     uint8_t * buff = (uint8_t *) malloc(buff_size);
 
-    if(buff) {
-        // read all data from server
-        while(connected() && (len > 0 || len == -1)) {
-
-            // get available data size
-            size_t sizeAvailable = _client->available();
-
-            if(sizeAvailable) {
-
-                int readBytes = sizeAvailable;
-
-                // read only the asked bytes
-                if(len > 0 && readBytes > len) {
-                    readBytes = len;
-                }
+    if(!buff) {
+        DEBUG_HTTPCLIENT("[HTTP-Client][writeToStreamDataBlock] not enough ram! need %d\n", HTTP_TCP_BUFFER_SIZE);
+        return HTTPC_ERROR_TOO_LESS_RAM;
+    }
 
-                // not read more the buffer can handle
-                if(readBytes > buff_size) {
-                    readBytes = buff_size;
-                }
+    // read all data from server
+    while(connected() && (len > 0 || len == -1))
+    {
+        int readBytes = len;
 
-                // read data
-                int bytesRead = _client->readBytes(buff, readBytes);
+        // not read more the buffer can handle
+        if(readBytes > buff_size) {
+            readBytes = buff_size;
+        }
 
-                // write it to Stream
-                int bytesWrite = stream->write(buff, bytesRead);
-                bytesWritten += bytesWrite;
+        // read data
+        int bytesRead = _client->readBytes(buff, readBytes);
+        if (!bytesRead)
+        {
+            DEBUG_HTTPCLIENT("[HTTP-Client][writeToStreamDataBlock] input stream timeout\n");
+            free(buff);
```

Comment:
Consider using above:
```cpp
std::unique_ptr<uint8_t[]> buff(new uint8_t[buff_size]);
```
above instead of the malloc, and removing the three free(buff) calls